cmd/juju/caas/add.go

```diff
@@ -499,7 +499,7 @@ func (c *AddCAASCommand) validateCloudRegion(cloudRegion string) (_ string, err
 }
 
 func (c *AddCAASCommand) getClusterMetadataFunc(ctx *cmd.Context) provider.GetClusterMetadataFunc {
-	return func(broker caas.ClusterMetadataChecker) (*caas.ClusterMetadata, error) {
+	return func(storageClassName string, broker caas.ClusterMetadataChecker) (*caas.ClusterMetadata, error) {
```

Comment:
This doesn't quite look right - we pass in storageClassName here but then below use c.workloadStorage and ignore the function param.
Let's discuss.